tokenisation.py

```diff
@@ -38,6 +132,116 @@
+                doc_treated = request[i-1]
+            documents += doc_treated
+            index_treated=[i-1,i]
+        if(request[i+1]=='NOT'):
+            index_treated.extend((i+1,i+2))
+            forbidden_doc+=inverted_index[request[i+2]]
+        elif(request[i+1]!='NOT'):
+            index_treated.append(i+1)
+            if type(request[i+1])==str:
+                doc_treated= inverted_index[request[i + 1]]
+            else:
+                doc_treated = request[i+1]
+            if(documents != []):
+                documents=intersection(documents,doc_treated)
+            else:
+                documents+=doc_treated
+        if len(forbidden_doc) > 0: # Removing forbidden docs from results
+            print('forbidden docs',forbidden_doc)
+            print('documents before removal', documents)
+            if len(documents) == 0:
+                documents = all_doc_id.copy()
+
+            for docId in documents:
+                if docId in forbidden_doc:
+                    print('docID', docId)
+                    documents.remove(docId)
+        request.insert(index_treated[0],documents)
+        for j in range(len(index_treated)):
+            request.pop(index_treated[1]) #
+        print('documents after removal', documents, ' and request is now :', request)
+        final_docs=documents
+    #Handling OR elements
+    while(next((operator for operator in request if operator == 'OR'), False)):
+        index_treated = []
+        forbidden_doc = []
+        i = request.index('OR')
+        if(request[i-2]=='NOT'):
+            index_treated =[i-2,i-1,i]
+            all_docs = all_doc_id.copy()
+            for docId in inverted_index[request[i - 1]]:
+                all_docs.remove(docId)
+            documents+= all_docs
+        else:
+            if type(request[i-1])==str:
+                doc_treated= inverted_index[request[i - 1]]
+            else:
+                doc_treated = request[i-1]
+            documents += doc_treated
+            index_treated=[i-1,i]
+        if(request[i+1]=='NOT'):
+            index_treated.extend((i+1,i+2))
+            all_doc = all_doc_id.copy()
+            for docId in inverted_index[request[i + 2]]:
+                all_doc.remove(docId)
+            documents+= all_doc
+        else:
+            index_treated.append(i+1)
+            if type(request[i+1])==str:
+                doc_treated= inverted_index[request[i + 1]]
+            else:
+                doc_treated = request[i+1]
+            documents+=doc_treated
+        request.insert(index_treated[0],documents)
+        for j in range(len(index_treated)):
+            request.pop(index_treated[1])
+        final_docs=documents
+    return list(set(final_docs))
+
+'''example of request'''
+
+print('mechanical : ',inverted_index['mechanical'])
+print('pragmatics :', inverted_index['pragmatics'])
+print('police : ', inverted_index['police'])
+request_andNot = 'pragmatics AND NOT pragmatics AND police'
```

Comment:
This peticular request for instance returns [] instead of inverted_index['police'] as it should theoretically

Reply:
resolved :)